controller/src/main/java/io/pravega/controller/store/stream/ZKStreamMetadataStore.java

```diff
@@ -460,6 +461,38 @@ public void unregisterBucketListener(int bucket) {
                           });
     }
 
+    /**
+     * When managing the Controller's metadata in Zookeeper, we explicitly create parent bucket zNodes (so they are of
+     * type "zNode"). Otherwise, they may be inadvertently created as Zookeeper "containers" by Curator. This would lead
+     * these zNodes to be candidates for automatic removal by Zookeeper if they become empty. For more information on
+     * the use of Zookeeper containers in Curator recipes, visit: https://curator.apache.org/curator-recipes/index.html
+     */
+    @Override
+    public CompletableFuture<Void> createBucketsRoot() {
+        List<CompletableFuture<Void>> initializationFutures = new ArrayList<>();
+        initializationFutures.add(initializeZNode(BUCKET_OWNERSHIP_PATH)); // Also initialize ownership zNode.
```

Comment:
please also add the root path explicitly as well.. 
following should be done:
"/buckets"
"/buckets/ownership"
"buckets/<bucket-id>"

Reply:
we also want to chain `bucketRootPath --> bucketOwnershipPath --> ListOf bucketPath` rather than run them concurrently as bucket path creation may fail if bucketRootPath is not yet created.